store/deposit-dates.js

```diff
@@ -46,9 +42,9 @@ class DepositDates {
     return Math.round(level * 100) / 100
   }
 
-  retrieveDepositDates = async (pageNumber, searchTerm, columnOrder) => {
+  async retrieveDepositDates(pageNumber, searchTerm, columnOrder) {
     const order = getOrder(columnOrder)
-    const key = `${this.rootStore.dataProvider}-${pageNumber}-${searchTerm}-${order}`
+    const key = `${pageNumber}-${searchTerm}-${order}`
```

Comment:
I don't think this was a good move. If you switch dataProvider you will see wrong data.

Reply:
No. I recreate child stores when data provider changes.